### ISSUE TITLE
No targets for training

### ISSUE THREAD

**Issue by abhinavkcs11** [microsoft/TAP#13]
When I train the VQA model I get the warning "Sample list has not field 'targets', are you sure that your ImDB has labels? you may have wanted to run with --evalai_inference 1"

I executed the same command as mentioned:
`python -m torch.distributed.launch --nproc_per_node 4 tools/run.py --tasks vqa --datasets m4c_textvqa --model m4c_split --seed 13 --config configs/vqa/m4c_textvqa/tap_refine.yml --save_dir save/m4c_split_refine_test --resume_file save/pretrained/textvqa_tap_base_pretrain.ckpt training_parameters.distributed True`

Can you provide additional details on this and how to train the model with the targets? And can you point out where the targets and the predictions are getting compared to compute loss?

**Comment by zyang-ur** [comment]
Hi @abhinavkcs11 ,

The loss function is here: https://github.com/microsoft/TAP/blob/main/pythia/modules/losses.py. 

Maybe we could trace from the "sample_list" in the loss functions, back to the data loader (e.g., https://github.com/microsoft/TAP/blob/352891f93c75ac5d6b9ba141bbe831477dcdd807/pythia/datasets/vqa/m4c_textvqa/dataset.py#L296) and see which part is not working as expected? Thank you.

**Comment by leonodelee** [comment]
> When I train the VQA model I get the warning "Sample list has not field 'targets', are you sure that your ImDB has labels? you may have wanted to run with --evalai_inference 1"
> 
> I executed the same command as mentioned: `python -m torch.distributed.launch --nproc_per_node 4 tools/run.py --tasks vqa --datasets m4c_textvqa --model m4c_split --seed 13 --config configs/vqa/m4c_textvqa/tap_refine.yml --save_dir save/m4c_split_refine_test --resume_file save/pretrained/textvqa_tap_base_pretrain.ckpt training_parameters.distributed True`
> 
> Can you provide additional details on this and how to train the model with the targets? And can you point out where the targets and the predictions are getting compared to compute loss?

Have you solved this problem?

**Comment by leonodelee** [comment]
> Hi @abhinavkcs11 ,
> 
> The loss function is here: https://github.com/microsoft/TAP/blob/main/pythia/modules/losses.py.
> 
> Maybe we could trace from the "sample_list" in the loss functions, back to the data loader (e.g.,
> 
> https://github.com/microsoft/TAP/blob/352891f93c75ac5d6b9ba141bbe831477dcdd807/pythia/datasets/vqa/m4c_textvqa/dataset.py#L296
> 
> ) and see which part is not working as expected? Thank you.

Thank you for your excellent work.
I got a similar problem when I train Textcaps model, there is no "target"in sample_info in textcaps dataset.py as you know, so what I should do to fix this?
